synapse/replication/tcp/resource.py

```diff
@@ -158,6 +158,21 @@ async def _run_notifier_loop(self) -> None:
                         all_streams = list(all_streams)
                         random.shuffle(all_streams)
 
+                    if self.command_handler.should_announce_positions():
+                        # We need to send out POSITIONs for all streams, usually
+                        # because a worker has reconnected.
+                        self.command_handler.will_should_announce_positions()
```

Comment:
`will_should_announce_positions` sets `_should_announce_positions` to false, so we're essentially saying that we've now announced positions, don't do it again next time around the loop?

Is it possible for another `REPLICATE` to come in after this? I guess that's fine, we'll just send them again next loop?

Reply:
Yup, pretty much. 

> so we're essentially saying that we've now announced positions, don't do it again next time around the loop?

We say we are *about* to send positions (to handle the race where something asks us to send out positions after we've just sent them out but before we clear the flag)